torch/_weights_only_unpickler.py

```diff
@@ -159,6 +159,11 @@ def _tensor_rebuild_functions():
 # Unpickling machinery
 @_functools.lru_cache(maxsize=1)
 def _get_allowed_globals():
+    dtensor_module = (
```

Comment:
@mikaylagawarecki 

Just FYI. I made some changes after your review. Since `torch.distributed.tensor` was made public not long ago, I had to add this check for backward compatibility to see whether pytorch is built with `torch.distributed.tensor` or `torch.distributed._tensor`.

Reply:
This is weird, I wouldn't expect us to need a BC check here

it seems like the issue is that mps does not have either `torch.distributed.tensor` or `torch.distributed._tensor`

Reply:
Ah you are right! macos doesn't have distributed built (use_distribute=0 for macos). Let me gate the additions as you suggested.